### ISSUE TITLE
nvme/zero_stage_3/optimizer/rank0/140354038028384.tensor.swp: buffer nbytes != file bytes 241969152 != 241958912

### ISSUE THREAD

**Issue by Deemo-cqs** [huggingface/transformers#25816]
### System Info

Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 4.32.0.dev0
- Platform: Linux-3.10.0-693.el7.x86_64-x86_64-with-glibc2.17
- Python version: 3.8.17
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.1
- Accelerate version: 0.22.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Information

- [x] The official example scripts
- [ ] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

deepspeed --num_gpus=1 run_translation.py \
--cache_dir /home/wangzhigangcs/bigmodelresearch/Deepspeed/DeepSpeedExamples/transformers/examples/pytorch/translation/mymodel/ \
--deepspeed /home/wangzhigangcs/bigmodelresearch/Deepspeed/DeepSpeedExamples/transformers/examples/pytorch/translation/config/ds_config_zero3.json \
--model_name_or_path t5-small --per_device_train_batch_size 4 \
--output_dir /home/wangzhigangcs/bigmodelresearch/Deepspeed/DeepSpeedExamples/transformers/examples/pytorch/translation/mymodel/ --overwrite_output_dir --fp16 \
--do_train --max_train_samples 300 --num_train_epochs 1 \
--dataset_name wmt16 --dataset_config "ro-en" \
--source_lang en --target_lang ro

{
    "fp16": {
        "enabled": true,
        "loss_scale": 0,
        "loss_scale_window": 1000,
        "initial_scale_power": 16,
        "hysteresis": 2,
        "min_loss_scale": 1
    },

    "bf16": {
        "enabled": "auto"
    },

    "optimizer": {
        "type": "AdamW",
        "params": {
            "lr": "auto",
            "betas": "auto",
            "eps": "auto",
            "weight_decay": "auto"
        }
    },

    "scheduler": {
        "type": "WarmupLR",
        "params": {
            "warmup_min_lr": "auto",
            "warmup_max_lr": "auto",
            "warmup_num_steps": "auto"
        }
    },

    "zero_optimization": {
        "stage": 3,
        "offload_optimizer": {
            "device": "nvme",
            "nvme_path": "/home/wangzhigangcs/bigmodelresearch/Deepspeed/DeepSpeedExamples/transformers/examples/pytorch/translation/nvme",
            "pin_memory": true,
            "buffer_count": 4,
            "fast_init": false
        },
        "offload_param": {
            "device": "cpu",
            "pin_memory": true
        },
        "overlap_comm": true,
        "contiguous_gradients": true,
        "sub_group_size": 1e9,
        "reduce_bucket_size": "auto",
        "stage3_prefetch_bucket_size": "auto",
        "stage3_param_persistence_threshold": "auto",
        "stage3_max_live_parameters": 1e9,
        "stage3_max_reuse_distance": 1e9,
        "stage3_gather_16bit_weights_on_model_save": true
    },
    "aio": {
        "block_size": 262144, 
        "queue_depth": 32, 
        "thread_count": 1, 
        "single_submit": false, 
        "overlap_events": true
    },

    "gradient_accumulation_steps": "auto",
    "gradient_clipping": "auto",
    "steps_per_print": 2000,
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "wall_clock_breakdown": false
}

### Expected behavior

Successfully offload into nvme

**Comment by amyeroberts** [comment]
cc @pacman100

**Comment by Deemo-cqs** [comment]
if i offload param into nvme and optimizer not, the program can run
